tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EmailSendMethod.scala

```diff
@@ -0,0 +1,240 @@
+      .flatMap(any => any)
+      .subscribeOn(Schedulers.elastic())
+
+  private def createEach(clientId: EmailSendCreationId,
+                         jsObject: JsObject,
+                         mailboxSession: MailboxSession,
+                         processingContext: ProcessingContext): SMono[(EmailSendResults, ProcessingContext)] = {
+    parseCreationRequest(jsObject)
+      .fold(error => SMono.error(error),
+        createEmailAndEmailSubmission(clientId, mailboxSession, _))
+      .map(response => EmailSendResults.created(clientId, response) -> processingContext)
+      .onErrorResume(error => SMono.just(EmailSendResults.notCreated(clientId, error) -> processingContext))
+  }
+
+  private def createEmailAndEmailSubmission(clientId: EmailSendCreationId,
+                                            mailboxSession: MailboxSession,
+                                            request: EmailSendCreationRequest): SMono[EmailSendCreationResponse] =
+    createEmail(clientId, mailboxSession, request.emailCreate)
+      .flatMap {
+        case failure: EmailSetCreationFailure => SMono.error(failure.error)
+        case success: EmailSetCreationSuccess => createEmailSubmission(mailboxSession, success.response, request.emailSubmissionSet)
+      }
+
+  private def parseCreationRequest(jsObject: JsObject): Either[Exception, EmailSendCreationRequest] =
+  EmailSendCreationRequest.validateProperties(jsObject)
+    .flatMap(validJson => serializer.deserializeEmailSendCreationRequest(validJson) match {
+      case JsSuccess(createRequest, _) => Right(createRequest)
+      case JsError(errors) => Left(EmailSendCreationRequestInvalidException.parse(errors))
+    })
+
+  def createEmail(clientId: EmailSendCreationId,
+                  mailboxSession: MailboxSession,
+                  request: EmailCreationRequest): SMono[EmailSetCreationResult] = {
+    val mailboxIds: List[MailboxId] = request.mailboxIds.value
+    if (mailboxIds.size != 1) {
+      SMono.just(EmailSetCreationFailure(clientId, new IllegalArgumentException("mailboxIds need to have size 1")))
+    } else {
+      SMono.fromCallable(() => request.toMime4JMessage(blobResolvers, htmlTextExtractor, mailboxSession))
+        .flatMap(either => either.fold(error => SMono.just(EmailSetCreationFailure(clientId, error)),
+          message => append(clientId, request, message, mailboxSession, mailboxIds)))
+        .onErrorResume(e => SMono.just[EmailSetCreationResult](EmailSetCreationFailure(clientId, e)))
+        .subscribeOn(Schedulers.elastic())
+    }
+  }
+
+  def createEmailSubmission(mailboxSession: MailboxSession,
+                            emailCreationResponse: EmailCreationResponse,
+                            request: EmailSubmissionCreationRequest): SMono[EmailSendCreationResponse] = {
+    val emailId: MessageId = emailCreationResponse.id
+    for {
+      message <- SFlux(messageIdManager.getMessagesReactive(List(emailId).asJava, FetchGroup.FULL_CONTENT, mailboxSession))
```

Comment:
Here we should cary over the mime message we have previously created, and NOT do a mailbox read (email in the mailbox would already be encrypted.)